### PR TITLE
Twitch Clips GET API Features

### PR DESCRIPTION
**Pull request by jaredcat**
A bunch of features based on twitch releasing their GET API endpoint for clips like resolving of the clip title, creator, and checking the broadcaster id (for filtering clips)
Permission checking for mod, subs, or broadcaster.

* Update .gitignore to ignore .idea folder from intellij

* Use es6 let and const for variables

* Clean up twitch channels initialization

* Add mod and subs only flags that can be turned on to limit to those who have those permissions

* Setup prettier, eslint, and autoformatting via husky

* Run prettier on js files

* Install dotenv to read from .env files

* Read from .env files for non-production server and add an example env file

* Add some debug logging output

* Add some comments, fix backward error log in post response, use === for booleans, switch anonymous functions to es6 notation

* Parse the URL slug from posted twitch clips to be used in API calls

* Enhanced data if a Twitch Client ID is provided. Resolve the name of the person who clipped, the title, and the game. Also broadcaster only flag

* Update readme and remove setting constants in index.js. Everything should be set as an environment variable (prod) or .env file (dev/debugging)

* Update README.md

* Only clips from restricted channels will be shared by default

* Run prettier on markdown

* Update readme since clive only sends clips from twitch channel list

### CODE REVIEW

**Comment by jaredcat** [comment]
Also added duplicated clip tracking so multiples of the same clip aren't posted

File: .env-example [hunk 1]
```diff
@@ -0,0 +1,11 @@
+NODE_ENV=dev
+LOG_LEVEL=debug
+LOG_FILE=/var/log/clive.log
+DB_FILE=/srv/clive/db.json
+DISCORD_WEBHOOK_URL=https://discordapp.com/api/webhooks/YOUR_DISCORD/WEBHOOK_URL_HERE
+TWITCH_CHANNELS="a_twitch_channel another_twitch_channel"
+TWITCH_CLIENT_ID=null
+RESTRICT_CHANNELS=trues
```
Comment (mangosango) [inline .env-example]: Should be `true` instead of `trues` (?)

**Review by mangosango — DISMISSED** [review]
Looks good aside from a possible typo!

File: .env-example (hunk 1 above)
Comment (jaredcat) [reply .env-example]: 👍

**Review by mangosango — APPROVED** [review]

**Comment by mangosango** [comment]
🎉

**Pull request merged**